app/auth/authV2/actions/RetrieveClientData.scala

```diff
@@ -47,23 +49,29 @@ class RetrieveClientData @Inject()(sessionDataService: SessionDataService,
 
     val useSessionDataService = appConfig.isSessionDataStorageEnabled
 
-    sessionDataService.getSessionData(useCookie = !useSessionDataService).map {
-      case Right(sessionData) => Right(ClientDataRequest(
-        sessionData.mtditid,
-        r.session.get(SessionKeys.clientFirstName),
-        r.session.get(SessionKeys.clientLastName),
-        sessionData.nino,
-        sessionData.utr,
-        getBooleanFromSession(SessionKeys.isSupportingAgent),
-        confirmed = {
-          r.session.get(SessionKeys.confirmedClient) match {
-            case Some(value) if value != "" => value.toBoolean
-            case _ => useSessionDataService
-          }
+    sessionDataService.getSessionData(useCookie = !useSessionDataService).flatMap {
```

Comment:
:thinking: - ideally if we can refactor this change to use for-comprehension, so this code would be flexible for changes in the future / EitherT / OptionT use really up to you to decide;